### PR TITLE
Features matching game leaderboard

### PR DESCRIPTION
**Pull request by ChocoMeow**

### CODE REVIEW

File: cogs/profile.py [hunk 1]
```diff
@@ -49,7 +49,8 @@ async def setbio(self, ctx: commands.Context, *, bio: str = None):
         """Sets your profile bio"""
         if bio and len(bio) > 30:
             return await ctx.reply(content="Please shorten the bio as it is too long. (No more than 30 chars)")
-
+        bio = bio.replace("`", "")
```
[the remark this answers is not in the document]
Comment (Arieswaran) [reply cogs/profile.py:52]: Only for bio? not for collection names? or card tags? btw do we need this sanitizing? As for as I know ,these queries doesn't work in pymongo.

File: cogs/info.py [hunk 2]
```diff
@@ -31,6 +31,29 @@ async def leaderboard(self, ctx: commands.Context):
...
 
+    @leaderboard.command(aliases=["mg"])
+    async def matchgame(self, ctx: commands.Context, level: str = "1"):
+        """Shows the IUFI Matching Game leaderboard."""
+        users = await func.USERS_DB.find().sort([
+            (f"game_state.match_game.{level}.matched", -1),
+            (f"game_state.match_game.{level}.click_left", -1),
+            (f"game_state.match_game.{level}.finished_time", 1)
+        ]).limit(10).to_list(10)
+
```
Comment (Arieswaran) [inline cogs/info.py:42]: Add a level check condition, check if that level exists first

File: cogs/profile.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ChocoMeow) [reply cogs/profile.py:52]: yes, i will also add that into collection name and card tag

File: cogs/info.py (hunk 2 above)
Comment (ChocoMeow) [reply cogs/info.py:42]: oh yes i forgot to pull i have that in testing code

**Review by Arieswaran — APPROVED** [review]

**Pull request merged**